spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala

```diff
@@ -200,6 +200,20 @@ abstract class AbstractModule[A <: Activity: ClassTag, B <: Activity: ClassTag,
   }
 
   /**
+   * whether this layer is trainable
+   */
+  var trainable: Boolean = true
+
+  /**
+   * set trainable
+   * @param trainable whether this layer is trainable
+   */
+  def setTrainable(trainable: Boolean): this.type = {
```

Comment:
The default value of trainable in layer should be true, since they need perform updateGradInput to support the whole back propagation link